packages/cache/src/cache.ts

```diff
@@ -459,6 +462,13 @@ async function saveCacheV2(
   options?: UploadOptions,
   enableCrossOsArchive = false
 ): Promise<number> {
+  // Override UploadOptions to force the use of Azure
+  options = {
+    ...options,
+    uploadChunkSize: 64 * 1024 * 1024, // 64 MiB
+    uploadConcurrency: 8, // 8 workers for parallel upload
+    useAzureSdk: true
+  }
```

Comment:
I wonder if we should flip the order here to allow `uploadChunkSize` and `uploadConcurrency` to be set by the caller rather than always overriding them.

```suggestion
  options = {
    uploadChunkSize: 64 * 1024 * 1024, // 64 MiB
    uploadConcurrency: 8, // 8 workers for parallel upload
    ...options,
    useAzureSdk: true
  }
```

Reply:
If we do that, they'll take on the default values set in `UploadOptions`. I'm explicitly overriding those defaults here. Once we remove the legacy service, we'll default to these values instead.

They can be overrode with the env variables here: https://github.com/actions/toolkit/pull/1882#discussion_r1866074795